next/frontend/utils/general.ts

```diff
@@ -92,3 +94,24 @@ export const taxStatusHelper = (tax: Tax) => {
   const hasMultipleInstallments = tax?.taxInstallments?.length > 1
   return { paymentStatus, hasMultipleInstallments }
 }
+
+// returns null if path is invalid or not an allowed origin
+export const getValidRedirectFromQuery = (path: unknown) => {
+  if (path && typeof path === 'string') {
+    // check if it's a local route
+    if (path.startsWith('/')) {
+      // TODO decodeURIComponent used originally, check if it's needed for some edge-case
+      return decodeURIComponent(path)
```

Comment:
@kristianduda wanted to check just this part with you - is this needed ? it was in the original redirect code but the next router hook seems to decode the uri for you - is there still reason to run it through decodeURICOmponent or can it be safely discarded ?